src/components/SendNativeToken/SendNativeTokenUI.tsx

```diff
@@ -0,0 +1,178 @@
+  useEffect(() => {
+    const triggerElementRef = triggerElement as React.MutableRefObject<HTMLElement> | undefined;
+    if (triggerElementRef && triggerElementRef.current) {
+      triggerElementRef.current.addEventListener('click', estimateAndSend);
+    }
+
+    // Cleanup
+    return () => {
+      if (triggerElementRef && triggerElementRef.current) {
+        triggerElementRef.current.removeEventListener('click', estimateAndSend);
+      }
+    };
+  }, [triggerElement]);
+
+  // Effect to validate the receiver address
+  useEffect(() => {
+    if (isValidEthereumAddress(receiverAddress)) {
+      setIsValidAddress(true);
+    } else {
+      setIsValidAddress(false);
+    }
+  }, [receiverAddress]);
+
+  // Handling key press events, specifically Enter key, for sending transactions
+  const handleEnterPress = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (!disableSendOnEnter && event.key === 'Enter') {
+      estimateAndSend();
+    }
+  };
+
+  // Handling input change events for the amount
+  const handleAmountChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setAmount(event.target.value);
+  };
+
+  // Logic for estimating and sending transactions
+  const estimateAndSend = async () => {
+    try {
+      if (isValidAddress) {
+        // Estimate logic
+        const gasEstimation = await estimate();
+  
+        if (!onlyEstimate) {
+          // Send logic
+          await send();
+          console.log('Transaction sent successfully');
+        }
+        // Log gas estimation result
+        console.log('Gas estimation result:', gasEstimation);
+      } else {
+        setError('Receiver address is not a valid blockchain address. Please double-check and try again.');
+      }
+    } catch (error) {
+      if (debug) {
+        console.error('Error in estimateAndSend:', error);
+  
+        if (error instanceof SomeSpecificEstimationError) {
+          console.error('Specific error in gas estimation:', error);
+        } else if (error instanceof SomeSpecificSendError) {
+          console.error('Specific error in sending transaction:', error);
+        }
+      }
+  
+      setError(
+        `An error occurred during the transaction. Please try again.${
+          debug ? ' Check the console for more details.' : ''
+        } If the issue persists, refer to the Etherspot documentation: https://etherspot.dev/docs`
+      );
+    }
+  };
+
+  // Rendering the UI component
+  return (
+    <EtherspotBatches>
+      <EtherspotBatch>
+        <EtherspotTransaction to={receiverAddress} value={amount}>
+          <div className="mt-2.5">
+            <input
+              type="text"
+              value={amount}
+              onChange={handleAmountChange}
+              onKeyDown={handleEnterPress}
+              placeholder="Enter Amount"
+              className={unstyled ? '' : className}
+              style={style}
+            />
+            {error ? <p className='text-sm'>{error}</p> : ''}
```

Comment:
No class names to define style, the developer should decide this. We'll need to devise a better system for this in the next iteration, but for now because you've added another HTML component here, we need to cater for a developer wanting to style this. 

- Please add another prop to the component called `errorMessageClass` and make it available to this HTML element to allow a developer to be able to style it.